compiler/src/dotty/tools/dotc/typer/VarianceChecker.scala

```diff
@@ -166,7 +166,19 @@ class VarianceChecker(using Context) {
   private object Traverser extends TreeTraverser {
     def checkVariance(sym: Symbol, pos: SrcPos) = Validator.validateDefinition(sym) match {
       case Some(VarianceError(tvar, required)) =>
-        def msg = i"${varianceLabel(tvar.flags)} $tvar occurs in ${varianceLabel(required)} position in type ${sym.info} of $sym"
+        def msg =
+          val enumAddendum =
+            val towner = tvar.owner
+            if towner.isAllOf(EnumCase) && towner.isClass
+              && tvar.span.exists && towner.span.exists
+              && tvar.span.start < towner.span.start // implies that `tvar` was not user declared
+            then
+              val example =
+                "See an example at http://dotty.epfl.ch/docs/reference/enums/adts.html#parameter-variance-of-enums"
+              i"\nenum case ${towner} requires explicit declaration of $tvar to resolve this issue.\n$example"
```

Comment:
This prints as "enum case class" which is a bit weird, we can use towner.name to avoid that:
```suggestion
              i"\nenum case ${towner.name} requires explicit declaration of $tvar to resolve this issue.\n$example"
```

Reply:
makes sense :)